pkg/gitops_test.go

```diff
@@ -530,19 +532,30 @@ func TestCloneGenerateAndPush(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			e := testutils.NewMockExecutor(tt.outputs...)
-			e.Errors = tt.errors
-			err := CloneGenerateAndPush(outputPath, repo, tt.component, e, tt.fs, "main", "/", true)
+
+			outputStack := testutils.NewOutputs(tt.outputs...)
+			executedCmds := []testutils.Execution{}
+			execute = func(baseDir string, cmd CommandType, args ...string) ([]byte, error) {
+				var output []byte
+				var execErr error
+				output, execErr, executedCmds = mockExecute(outputStack, tt.errors, executedCmds, baseDir, cmd, args...)
+				return output, execErr
+			}
```

Comment:
It seems like we're doing this in every test case..

I would just do this as a implementation of an interface once (can replace the Executor interface in gitops.go). So we can 2 implementation:
1. original execute
2. test execute

this will also solve the issue of not resetting execute at the end of every test 🤔

Reply:
This is a good suggestion (I don't like duplicate code either) but the limiting factor here is, the parameters to the mockExecute call varies.  I can't pass in test arguments like `outputStack, tt.errors, executedCmds` to the interface function because that has to stay fixed in order for the override to work, so I have to inline the function declaration in each test in order to get the values of the test arguments that are set outside the scope of the execute function.

Reply:
I think we could also put most of these in a util getter func? Does this work? But I leave it up to you

```
execute = newTestExecute(outputStack, tt.errors, executedCmds, baseDir, cmd, args...)

func newTestExecute(outputStack, tt.errors, executedCmds, baseDir, cmd, args...) func {
    return func(baseDir string, cmd CommandType, args ...string) ([]byte, error) {
				var output []byte
				var execErr error
				output, execErr, executedCmds = mockExecute(outputStack, tt.errors, executedCmds, baseDir, cmd, args...)
				return output, execErr
			}
}
```

Reply:
Yes, this does work.  It's a lot cleaner so I will implement this.  Thanks for the suggestions!